packages/glimmer-runtime/lib/compiled/expressions/concat.ts

```diff
@@ -31,10 +31,28 @@ class ConcatReference extends CachedReference<string> {
   }
 
   protected compute(): string {
-    let parts = new Array<string>(this.parts.length);
+    let parts = new Array<string>();
+
     for (let i = 0; i < this.parts.length; i++) {
-      parts[i] = normalizeTextValue(this.parts[i].value());
+      let value = this.parts[i].value();
+
+      if (value !== null && value !== undefined) {
+        parts[i] = castToString(this.parts[i].value()) as FIXME<string, 'Coerce falsy values to strings'>;
```

Comment:
IIUC, if we didn't pre-size it (which has a shaping problem, as we discovered recently), we should use push to avoid going OOB, is that correct? /cc @krisselden